pycqed/measurement/measurement_control.py

```diff
@@ -269,6 +311,31 @@ def run(self, name: str=None, exp_metadata: dict=None,
             #  saved.
             self.save_timers(self.data_object)
             return_dict = self.create_experiment_result_dict()
+            if exception is not None:  # exception occurred in above try-block
+                if previous_attempts + 1 < self.max_attempts():
+                    # Maximum number of attempts not reached. Log to logger
+                    # and to slack, and retry.
+                    msg = f'MC: Error during measurement (attempt ' \
+                          f'{previous_attempts + 1} of {self.max_attempts()}' \
+                          f'). Retrying.'
+                    self.log_to_slack(msg)
+                    log.error(msg)
+                    log.error(formatted_exc)
+                    # Call the retry_cleanup_functions if there are any.
+                    [fnc() for fnc in getattr(
+                        self, 'retry_cleanup_functions', [])]
+                    # sweep points should be extracted again from the sweep
+                    # function to avoid tiling them multiple times (in every
+                    # attempt)
+                    del self.sweep_points
```

Comment:
wait isn't this taken care of in line 234?

Reply:
No, not this kind of tiling ;-). I have extended the comment.